yeonjae/programmers-2/괄호회전하기.js

```diff
@@ -0,0 +1,33 @@
+function solution(s) {
+  const bracketPair = { "(": ")", "[": "]", "{": "}" };
+  const bracketKeys = Object.keys(bracketPair);
+
+  return s.split("").reduce((acc, v, i) => {
+    const curStr = s.slice(i) + s.slice(0, i);
+    const stack = [];
+
+    for (const curChar of curStr) {
+      if (bracketKeys.includes(curChar)) {
+        stack.push(curChar);
+      } else {
+        const lastStr = stack.pop();
+
+        if (bracketPair[lastStr] !== curChar) return acc;
+      }
+    }
+
+    return stack.length ? acc : ++acc;
+  }, 0);
+}
```

Comment:
```suggestion
function solution(s) {
  const bracketPair = { "(": ")", "[": "]", "{": "}" };

  return s.split("").reduce((acc, v, i) => {
    const curBuckets = s.slice(i) + s.slice(0, i);
    const stack = [];
      
    for (const bucket of curBuckets) {
      if (bracketPair[bucket]) {
        stack.push(bucket);
      } else {
        const lastStr = stack.pop();

        if (bracketPair[lastStr] !== bucket) return acc;
      }
    }

    return stack.length ? acc : ++acc;
  }, 0);
}
```
- const curBuckets = s.slice(i) + s.slice(0, i); 부분 신박하네요! 좋은 풀이 잘봤습니다.
- 네이밍을 약간 변경해봤습니다 Str, curChar이런것보다 괄호라는 의미를 담아봤습니다~
- includes를 사용할 필요없이 bracketPair를 접근했을 때 value가 있냐 없냐로 판단할 수가 있어요 당연히 객체 조회로 판단하기 때문에 효율성도 높아집니다 👍